packages/ra-ui-materialui/src/layout/Notification.js

```diff
@@ -71,7 +71,7 @@ class Notification extends React.Component {
             hideNotification,
             ...rest
         } = this.props;
-
+        const { warning, confirm, undo, ...snackbarClasses } = classses;
```

Comment:
Typo here. It should be `classes` with only 2 `s`. You obviously didn't test your changes at all so I'm not comfortable with merging your PR.

Reply:
@djhi I just tested in example demo, there is also duplicate variable declaration "undo" in same render function scope. Is there any other solution to remove unknown classes? I could fix the scope issue by separating the logic to function. 
```
getSnackbarClasses = ({ warning, confirm, undo, ...snackbarClasses }) => snackbarClasses

// In render function
const snackbarClasses = this.getSnackbarClasses(classes);
```

Reply:
Yup, you can use the ability to rename a property when destructuring.

```js
const {
    warning,
    confirm,
    undo: undoClass, // Rename classes.undo to undoClass in this scope to avoid name conflicts
    ...snackbarClasses,
} = classes;
```

Reply:
Thanks, that's a nice solution. :smile: